incubator/binding-openapi/src/main/java/io/aklivity/zilla/runtime/binding/openapi/internal/config/OpenapiClientCompositeBindingAdapter.java

```diff
@@ -0,0 +1,345 @@
+    {
+        if (responses.headers != null && !responses.headers.isEmpty())
+        {
+            for (Map.Entry<String, Header> header : responses.headers.entrySet())
+            {
+                String name = header.getKey();
+                ModelConfig model = models.get(header.getValue().schema.type);
+                if (model != null)
+                {
+                    response
+                        .header()
+                            .name(name)
+                            .model(model)
+                            .build();
+                }
+            }
+        }
+        return response;
+    }
+
+    private <C> NamespaceConfigBuilder<C> injectTlsClient(
+        NamespaceConfigBuilder<C> namespace,
+        TlsOptionsConfig tlsConfig,
+        boolean secure)
+    {
+        if (secure)
+        {
+            namespace
+                .binding()
+                    .name("tls_client0")
+                    .type("tls")
+                    .kind(CLIENT)
+                    .options(tlsConfig)
+                    .vault("client")
+                    .exit("tcp_client0")
+                    .build();
+        }
+        return namespace;
+    }
+
+    private <C> NamespaceConfigBuilder<C> injectCatalog(
+        NamespaceConfigBuilder<C> namespace,
+        OpenApi openApi)
+    {
+        if (openApi.components != null &&
+            openApi.components.schemas != null &&
+            !openApi.components.schemas.isEmpty())
+        {
+            namespace
+                .catalog()
+                    .name(INLINE_CATALOG_NAME)
+                    .type(INLINE_CATALOG_TYPE)
+                    .options(InlineOptionsConfig::builder)
+                        .subjects()
+                            .inject(s -> this.injectSubjects(s, openApi))
+                            .build()
+                        .build()
+                    .build();
+        }
+        return namespace;
+    }
+
+    private <C> InlineSchemaConfigBuilder<C> injectSubjects(
+        InlineSchemaConfigBuilder<C> subjects,
+        OpenApi openApi)
+    {
+        try (Jsonb jsonb = JsonbBuilder.create())
+        {
+            YAMLMapper yaml = YAMLMapper.builder()
+                .disable(WRITE_DOC_START_MARKER)
+                .enable(MINIMIZE_QUOTES)
+                .build();
+            for (Map.Entry<String, Schema> entry : openApi.components.schemas.entrySet())
+            {
+                SchemaView schema = SchemaView.of(openApi.components.schemas, entry.getValue());
+                subjects
+                    .subject(entry.getKey())
+                        .version(VERSION_LATEST)
+                        .schema(writeSchemaYaml(jsonb, yaml, schema))
+                        .build();
+            }
+        }
+        catch (Exception ex)
+        {
+            rethrowUnchecked(ex);
+        }
+        return subjects;
+    }
+
+    private static String writeSchemaYaml(
```

Comment:
This can be removed, and if some dependencies are no longer needed then those can be removed too.

Reply:
I thought we need to catalog